### PR TITLE
New forms

### PR DESCRIPTION
**Pull request by AlyAtalla**
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post.likes.create(user: current_user)
+    redirect_to post_path(@post), notice: 'Post liked successfully.'
+  end
+
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [ ] Creation of like should not be implemented in the Posts controller. Like should have its own controller with its own `create` action. Kindly add these and remove like feature from the Posts controller.

File: config/routes.rb [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-# Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-# Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
-# Can be used by load balancers and uptime monitors to verify that the app is live.
-get "up" => "rails/health#show", as: :rails_health_check
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :create, :new] do
+      resources :comments, only: [:new, :create]
+    end
+  end
```
Comment (TedLivist) [inline config/routes.rb]: - [ ] There should be a `:create` Like route, just as you have the create comment route.

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
+  </div>
+  <h4 class="comment-title">Comments</h4>
+  <div class="post-body">
+  <%= @post.text %>
+</div>
+  <div class="comment-container">
+    <% @post.comments.each do |comment| %>
+      <span><%= comment.author.name if comment.author %>: <%= comment.text %></span>
+    <% end %>
+  </div>
```
Comment (TedLivist) [inline app/views/posts/show.html.erb]: - [ ] Kindly add some kind of button that allows the user to like a post, as there is currently no way for a user to add a like on your app.

  image

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,14 +1,45 @@
...
+  private
+
+  def set_user
+    @user = User.find(params[:user_id])
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [ ] The posts should be created on behalf of the `current_user`, not the user from the route. Kindly fix that.

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
+    @comment = Comment.new
   end
 
-  def show
-    @comment = Comment.find(params[:id])
-    render status: :ok
+  def create
+    @comment = Comment.new(comment_params)
+    @post = Post.find(params[:post_id])
+    @comment.author = User.find(params[:user_id])
```
Comment (TedLivist) [inline app/controllers/comments_controller.rb]: - [ ] The author of the comment should be the `current_user`, not user from the URL route

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @AlyAtalla,

Good job so far!
✔️ Posts and comments routes and creation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -11,4 +11,27 @@ def index
...
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = User.find(params[:user_id])
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [ ] @AlyAtalla Great work implementing the changes requested, however you have not implemented the change requested, kindly ensure the author of the post is the current_user and not the user from the routes.

File: app/controllers/comments_controller.rb (hunk 5 above)
Comment (danmainah) [reply app/controllers/comments_controller.rb]: - [ ] @AlyAtalla Great work implementing the changes requested, however you have not implemented the change requested, kindly ensure the author of the comment is the current_user and not the user from the routes.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AlyAtalla , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented the changes requested and added the Like controller and a button to add like to a post.
✔️ Added route for the like controller to ensure the like functionality works correctly.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @AlyAtalla , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️Implemented the change requested to ensure the posts and comments are created on behalf of the current user.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**